Add support for batched & persisted queries.

DFurnes opened the pull request:
This pull request adds support for [apollo-link-batch-http](https://www.apollographql.com/docs/link/links/batch-http/) and [apollo-link-persisted-queries](https://github.com/apollographql/apollo-link-persisted-queries). If an array of queries is provided, it will resolve each one and then return the array of responses.

If the client tries to use persisted queries (which don't include the query itself in the payload), it will return a `PersistedQueryNotSupported` error so that the request is retried as a standard query.

DFurnes committed:
Add support for 'apollo-link-batch-http'.

DFurnes committed:
Add support for 'apollo-link-persisted-queries'.

DFurnes commented:
If anyone wants to test this out, I’ve pushed a corresponding branch with the build directory included:

```json
"cypress-graphql-mock": "DFurnes/cypress-graphql-mock#with-persisted-batch-support",
```

tgriesser commented:
Hey @DFurnes - sorry for letting this sit for awhile. We now have proper cypress tests in this repo, so if you can just add a test for this feature & resolve the merge conflict, we can get it in there!

DFurnes committed:
Add test for persisted queries.

This required "ejecting" from Apollo Boost, which does not support customizing which links are used.

DFurnes committed:
Add test for batched queries.

DFurnes commented:
@tgriesser Rebased & added tests for each new feature in 1bfef61 and fa9b17f!

DFurnes commented:
Hi @tgriesser, just wanted to check in & see if we can get this reviewed and merged in! 😃 Let me know if there's any other changes or documentation that'd be helpful.

kj-brown commented:
Just going to bump this; hopefully it becomes available in the main repo!
